fineract-provider/src/main/java/org/apache/fineract/portfolio/account/service/AccountAssociationsReadPlatformServiceImpl.java

```diff
@@ -99,22 +99,22 @@ public boolean isLinkedWithAnyActiveAccount(final Long savingsId) {
 
         final List<Map<String, Object>> statusList = this.jdbcTemplate.queryForList(sql1, savingsId);
         for (final Map<String, Object> statusMap : statusList) {
-            AccountAssociationType associationType = AccountAssociationType.fromInt((Integer) statusMap.get("type"));
+            AccountAssociationType associationType = AccountAssociationType.fromInt(((Short) statusMap.get("type")).intValue());
```

Comment:
Why is it `short`?

Reply:
Because in the DB is it `short` or `smallint` and casting it to Integer would would sometimes raise ClassCastException on Ubuntu OS.

Reply:
Sometimes? What do you mean? From 5 calls, 4 times it is integer and 1 time it short and you got ClassCastException?

Reply:
I would say 1/ 5 the exception would occur.

Reply:
That sounds really weird...

Reply:
Yes, the weird part is the _sometimes_ bit. I did not manage to get a proper pattern. It would sometimes just work and sometimes it would raise the exception. But ever since I applied this fix I have not had the issue.